spec/fields_spec.rb

```diff
@@ -1,49 +1,53 @@
 require 'spec_helper'
 
 RSpec.describe 'fields' do
-  include_context 'scoping'
+  include JsonHelpers
+  include_context 'resource testing'
+  let(:resource) { Class.new(PORO::EmployeeResource).new }
+  let(:base_scope) { { type: :employees } }
 
-  class SerializableTestFields < JSONAPI::Serializable::Resource
-    type 'authors'
-
-    attribute :first_name
-    attribute :last_name
-    attribute :uuid do
-      SecureRandom.uuid
-    end
-    attribute :salary, if: proc { !@context || @context.current_user == 'admin' } do
-      50_000
-    end
-
-    def admin?
-      scope == 'admin'
-    end
+  let!(:employee) do
+    PORO::Employee.create(first_name: 'John', last_name: 'Doe')
   end
 
-  let!(:author) { Author.create!(first_name: 'Stephen', last_name: 'King') }
-
-  def json
-    render(scope.resolve, class: { Author: SerializableTestFields })
-  end
+  subject(:attributes) { json['data'][0]['attributes'] }
 
   it 'does not limit without fields param' do
-    expect(json['data'][0]['attributes'].keys).to eq(%w(first_name last_name uuid salary))
+    render
+    expect(attributes.keys).to eq(%w(first_name last_name age))
```

Comment:
Should add spec that verifies `salary` is returned if the user is admin but not specifying the fields.

Reply:
good call